pymc/data.py

```diff
@@ -218,9 +218,9 @@ def Minibatch(variable: TensorVariable, *variables: TensorVariable, batch_size:
 def determine_coords(
     model,
     value: pd.DataFrame | pd.Series | xr.DataArray,
-    dims: Sequence[str | None] | None = None,
```

Comment:
Did this ever change? I thought that some variables could have `dims=[None, "some_dim_name"]`. If this is still the case, then the type hint needs to left as it was before.

Reply:
I think you're right. I guess someone could pass a data cube with some "batch" or otherwise uninteresting dimension? I don't really see why you would ever do that, but I'm not gonna stop them. I'll change it back.

Reply:
Actually, do you mind giving an example? I observe that in main branch, both `pm.Data` and `pm.ConstantData` take `dims: Sequence[str] | None`. Are these incorrect and `Sequence[str]` should be `Sequence[str | None]`?

Reply:
I don't think we allow mixed None / defined dims anymore

Reply:
I know that it used to be possible in the past, but at some point @michaelosthege and @ricardoV94 discussed about simplifying the code base and removing part of this flexibility. I don't recall the details, but with what Ricardo commented, I think it's safe to assume that the flexibility here is gone and that your type hints are correct.